mediator/esmFldsExchange_hafs_mod.F90

```diff
@@ -186,40 +212,72 @@ subroutine esmFldsExchange_hafs_advt(gcomp, phase, rc)
     ! to ocn: state fields
     ! ---------------------------------------------------------------------
     if (hafs_attr%atm_present .and. hafs_attr%ocn_present) then
-      allocate(S_flds(6))
-      S_flds = (/'Sa_u10m', & ! inst_zonal_wind_height10m
-                 'Sa_v10m', & ! inst_merid_wind_height10m
-                 'Sa_t2m ', & ! inst_temp_height2m
-                 'Sa_q2m ', & ! inst_spec_humid_height2m
-                 'Sa_pslv', & ! inst_pres_height_surface
-                 'Sa_tskn' /) ! inst_temp_height_surface
-      do n = 1,size(S_flds)
-         fldname = trim(S_flds(n))
-         call addfld_from(compatm, trim(fldname))
-         call addfld_to(compocn, trim(fldname))
-      end do
-      deallocate(S_flds)
+       if (trim(coupling_mode) == 'hafs') then
+          allocate(S_flds(6))
+          S_flds = (/'Sa_u10m', & ! inst_zonal_wind_height10m
+                     'Sa_v10m', & ! inst_merid_wind_height10m
+                     'Sa_t2m ', & ! inst_temp_height2m
+                     'Sa_q2m ', & ! inst_spec_humid_height2m
+                     'Sa_pslv', & ! inst_pres_height_surface
+                     'Sa_tskn' /) ! inst_temp_height_surface
+          do n = 1,size(S_flds)
+             fldname = trim(S_flds(n))
+             call addfld_from(compatm, trim(fldname))
+             call addfld_to(compocn, trim(fldname))
+          end do
+          deallocate(S_flds)
+       else
+          allocate(S_flds(1))
+          S_flds = (/'Sa_pslv'/) ! inst_pres_height_surface
+          do n = 1,size(S_flds)
+             fldname = trim(S_flds(n))
+             call addfld_from(compatm, trim(fldname))
+             call addfld_to(compocn, trim(fldname))
+          end do
+          deallocate(S_flds)
+       end if
     end if
 
     ! ---------------------------------------------------------------------
     ! to ocn: flux fields
     ! ---------------------------------------------------------------------
     if (hafs_attr%atm_present .and. hafs_attr%ocn_present) then
-      allocate(F_flds(7,2))
-      F_flds(1,:) = (/'Faxa_taux ','Faxa_taux '/) ! mean_zonal_moment_flx_atm
-      F_flds(2,:) = (/'Faxa_tauy ','Faxa_tauy '/) ! mean_merid_moment_flx_atm
-      F_flds(3,:) = (/'Faxa_rain ','Faxa_rain '/) ! mean_prec_rate
-      F_flds(4,:) = (/'Faxa_swnet','Faxa_swnet'/) ! mean_net_sw_flx
-      F_flds(5,:) = (/'Faxa_lwnet','Faxa_lwnet'/) ! mean_net_lw_flx
-      F_flds(6,:) = (/'Faxa_sen  ','Faxa_sen  '/) ! mean_sensi_heat_flx
-      F_flds(7,:) = (/'Faxa_lat  ','Faxa_lat  '/) ! mean_laten_heat_flx
-      do n = 1,size(F_flds,1)
-         fldname1 = trim(F_flds(n,1))
-         fldname2 = trim(F_flds(n,2))
-         call addfld_from(compatm, trim(fldname1))
-         call addfld_to(compocn, trim(fldname2))
-      end do
-      deallocate(F_flds)
+       if (trim(coupling_mode) == 'hafs') then
+          allocate(F_flds(7,2))
+          F_flds(1,:) = (/'Faxa_taux ','Faxa_taux '/) ! mean_zonal_moment_flx_atm
```

Comment:
We need to update the comment description. All of the fields, even under just "hafs" are now inst fields.

Reply:
@DeniseWorthen done.